### PR TITLE
Fixes mining shuttle landing inside the aux base

### PR DESCRIPTION
**Pull request by GunHog**
The mining shuttle beacon now properly checks that none of the turfs of
the shuttle's landing zone intersect the Aux Base.

Fixes https://github.com/tgstation/tgstation/issues/24414

### CODE REVIEW

**Comment by Sligneris** [comment]
So, does this finally allow landing the shuttle _right_ next to aux base?

That issue had a funny way of not preventing shuttle colliding with the base when facing the wrong way, but forced me to summon the shuttle free tiles away from the base when facing the right way.

**Comment by GunHog** [comment]
Yes, that it does. I need a better sprite for the beacon, but if you point it away from the base, you can land the shuttle directly adjacent to one of your base's airlocks to use as a dock. Drag the beacon immediately outside your desired door, and make sure it is pointing to the outside.

File: code/modules/mining/aux_base.dm [hunk 1]
```diff
@@ -302,26 +302,30 @@ obj/docking_port/stationary/public_mining_dock/onShuttleMove()
 	if(!Mport)
 		user << "<span class='warning'>This station is not equipped with an approprite mining shuttle. Please contact Nanotrasen Support.</span>"
 		return
-	var/search_radius = max(Mport.width, Mport.height)*0.5
```
Comment (RemieRichards) [inline code/modules/mining/aux_base.dm]: >Mport

hilarious

File: code/modules/mining/aux_base.dm [hunk 2]
```diff
@@ -302,26 +302,30 @@ obj/docking_port/stationary/public_mining_dock/onShuttleMove()
...
 	if(!mining_shuttle) //Not having a mining shuttle is a map issue
 		user << "<span class='warning'>No mining shuttle signal detected. Please contact Nanotrasen Support.</span>"
 		SSshuttle.stationary.Remove(Mport)
 		qdel(Mport)
 		return
 
+	for(var/L in landing_turfs) //You land NEAR the base, not IN it.
+		if(istype(get_area(L), /area/shuttle/auxillary_base))
+			user << "<span class='warning'>The mining shuttle must not land within the mining base itself.</span>"
+			SSshuttle.stationary.Remove(Mport)
```
Comment (AnturK) [inline code/modules/mining/aux_base.dm]: Is there a reason this isn't a part of dock Destroy ?

Comment (GunHog) [inline code/modules/mining/aux_base.dm]: I do not know. I did not write shuttle code, I would rather not be the one to change it and risk breaking all shuttles or something.

Comment (AnturK) [inline code/modules/mining/aux_base.dm]: Alright, can be changed in another PR.
